### PR TITLE
Update Controller Runtime Go module to v0.8.2

### PR DESCRIPTION
**Pull request by bmelbourne**
**What this PR does / why we need it**:
Update sigs.k8s.io `controller-runtime` Go module to **v0.8.2**.

- sigs.k8s.io/controller-runtime v0.6.2 => v0.8.2
- github.com/go-logr/logr upgrade => v0.4.0
- github.com/gogo/protobuf upgrade => v1.3.2
- github.com/google/uuid upgrade => v1.2.0
- github.com/prometheus/client_golang upgrade => v1.9.0
- golang.org/x/crypto upgrade => v0.0.0-20210220033148-5ea612d1eb83
- golang.org/x/net upgrade => v0.0.0-20210222171744-9060382bd457
- golang.org/x/oauth2 upgrade => v0.0.0-20210220000619-9bb904979d93
- golang.org/x/sys upgrade => v0.0.0-20210223095934-7937bea0104d
- k8s.io/api upgrade => v0.20.4
- k8s.io/apimachinery upgrade => v0.20.4
- k8s.io/client-go upgrade => v1.5.1
- k8s.io/component-base upgrade => v0.20.4
- k8s.io/klog/v2 upgrade => v2.5.0
- k8s.io/utils upgrade => v0.0.0-20210111153108-fddb29f9d009

**Which issue(s) this PR fixes**:
N/A

**Special notes for your reviewer**:
None

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -91,26 +77,33 @@ require (
...
 	google.golang.org/api v0.40.0
 	gopkg.in/gcfg.v1 v1.2.3
 	gopkg.in/inf.v0 v0.9.1
+	gopkg.in/yaml.v2 v2.4.0 // indirect
+	gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b // indirect
 	helm.sh/helm/v3 v3.4.2
 	honnef.co/go/tools v0.0.1-2020.1.6 // indirect
-	k8s.io/api v0.20.0
-	k8s.io/apimachinery v0.20.0
+	k8s.io/api v0.20.4
```
Comment (olemarkus) [inline go.mod:93]: Maybe change the k8s.io replace above to v0.20.4? At least these changes will not take effect otherwise.

**Review by olemarkus — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (bmelbourne) [reply go.mod:93]: I was going to bump those up to `v0.20.4` in my next PR when I update kubectl to the latest version. I'm waiting for each PR to be merged first before raising the next due to some of the challenges around dependencies, etc.

**Review by olemarkus — APPROVED** [review]
Thanks!

**Pull request merged**